### DISCUSSION TITLE
Can spades be used to align small RNA data?

### DISCUSSION THREAD

**Discussion by abracarambar** [ablab/spades#786] [Q&A]
Hi, 
We are interested to assemble viruses from plants using sRNA libraries. We generally find one or several viruses within a plant. Most of the sequencing reads match to sRNA derived from the plant but a small fraction represents virus-derived siRNA. The reads we are interested in range from 21-24 nt in length. We have tested SPADES 3.12 and 3.14 with kmer size ranging from 9 to 21 but we observe some missasemblies. We also fail to recover viruses present at low titer.  Is there a setting and mode in the latest SPADES version (3.15) you recommend to use that would allow us to best assemble de novo these virus reads? And are earlier versions of SPADES possibly more geared for sRNA assemblies? Thanks for your feedback.

Example of command currently run with SPADES 3.12 and 3.14
`spades.py -t 2 -k 9-21 --only-assembler -m 180 -s $samplefile -o ${samplefile.baseName}_spades_${kmer}`

Warnings we get when we run the tools with this setting:
=== Error correction and assembling warnings:
 * 0:01:01.112   373M / 2G    WARN    General                 (kmer_coverage_model.cpp   : 366)   Failed to determine erroneous kmer threshold. Threshold set to: 60
 * 0:00:44.483   266M / 2G    WARN    General                 (kmer_coverage_model.cpp   : 218)   Too many erroneous kmers, the estimates might be unreliable
 * 0:00:51.928   287M / 2G    WARN    General                 (kmer_coverage_model.cpp   : 366)   Failed to determine erroneous kmer threshold. Threshold set to: 41
 * 0:00:34.433   348M / 733M  WARN    General                 (kmer_coverage_model.cpp   : 327)   Valley value was estimated improperly, reset to 2
 * 0:00:34.434   348M / 733M  WARN    General                 (kmer_coverage_model.cpp   : 366)   Failed to determine erroneous kmer threshold. Threshold set to: 2
 * 0:00:18.697   208M / 256M  WARN    General                 (kmer_coverage_model.cpp   : 366)   Failed to determine erroneous kmer threshold. Threshold set to: 22

**Comment by andrewprzh** [comment]
Hi

Unfortunately we have never tested our software on reads as short as 21-24 bp. Thus, I cannot promise sane quality of the assembly.
In any case, I'd suggest to
- Use either RNA or RNA Viral mode (--rna or --rnaviral respectively). That may save you some low abundant sequences and will handle non-uniform coverage much better.
- Do not use very small k values, e.g. I'd suggest to use k=19,21 (17,21 at minimum). Using k below 15 may create a lot of misassemblies, as you indicated. Even with k=19,21 you may still get a lot of them.

Best
Andrey
